benchmarks/sign.mjs

```diff
@@ -1,21 +1,16 @@
 'use strict'
 
-import { isMainThread } from 'worker_threads'
 import { privateKeys, publicKeys, compareSigning, saveLogs } from './utils.mjs'
 
-async function runSuites() {
-  if (!isMainThread) {
-    const algorightm = process.env.CURRENT_ALGORITHM
-    compareSigning({ a: 1, b: 2, c: 3 }, algorightm, privateKeys[algorightm], publicKeys[algorightm])
-    return
-  } else {
-    for (const algorightm of ['HS512', 'ES512', 'RS512', 'PS512', 'EdDSA']) {
-      process.env.CURRENT_ALGORITHM = algorightm
-      await compareSigning({ a: 1, b: 2, c: 3 }, algorightm, privateKeys[algorightm], publicKeys[algorightm])
-    }
+export async function runSuites() {
+  const benchmarkResults = []
+  for (const algorithm of ['HS512', 'ES512', 'RS512', 'PS512', 'EdDSA']) {
+    const result = await compareSigning({ a: 1, b: 2, c: 3 }, algorithm, privateKeys[algorithm], publicKeys[algorithm])
+    benchmarkResults.push({ algorithm, result })
   }
 
   await saveLogs('sign')
+  return benchmarkResults
 }
 
-runSuites().catch(console.error)
+if (import.meta.filename === process.argv[1]) runSuites().catch(console.error)
```

Comment:
if we keep `.catch`-ing errors in this way, are we not potentially hiding errors and therefore risk that they go unnoticed as they did already?